consultation_analyser/consultations/models.py

```diff
@@ -40,301 +41,237 @@ class Meta:
-                has_multiple_selections = cursor.fetchone()[0]
-
-            output.append(
-                MultipleChoiceQuestionStats(
-                    question=q,
-                    counts=counts,
-                    total_responses=total_responses,
-                    has_multiple_selections=has_multiple_selections,
-                )
-            )
 
-        return output
-
-    text = models.CharField(
-        null=False, max_length=None
-    )  # no idea what's a sensible value for max_length
-    slug = models.CharField(null=False, max_length=None)
-    has_free_text = models.BooleanField(default=False)
-    multiple_choice_options = models.JSONField(
-        null=True, blank=True, validators=[MultipleChoiceSchemaValidator(limit_value=None)]
-    )
-    section = models.ForeignKey(Section, on_delete=models.CASCADE)
+class Respondent(UUIDPrimaryKeyModel, TimeStampedModel):
+    consultation = models.ForeignKey(Consultation, on_delete=models.CASCADE)
+    # demographic data, or anything else that is at respondent level
+    data = models.JSONField(default=dict)
 
     class Meta(UUIDPrimaryKeyModel.Meta, TimeStampedModel.Meta):
-        constraints = [
-            models.UniqueConstraint(fields=["slug", "section"], name="unique_question_section"),
-        ]
+        pass
 
 
-class ConsultationResponse(UUIDPrimaryKeyModel, TimeStampedModel):
-    consultation = models.ForeignKey(Consultation, on_delete=models.CASCADE)
-    submitted_at = models.DateTimeField(editable=False, null=False)
+class Answer(UUIDPrimaryKeyModel, TimeStampedModel):
+    question_part = models.ForeignKey(QuestionPart, on_delete=models.CASCADE)
+    respondent = models.ForeignKey(Respondent, on_delete=models.CASCADE)
+    text = models.TextField()
+    chosen_options = models.JSONField(default=list)
+    # TODO - add favourite
 
     class Meta(UUIDPrimaryKeyModel.Meta, TimeStampedModel.Meta):
         pass
 
 
-class ProcessingRun(UUIDPrimaryKeyModel, TimeStampedModel):
-    consultation = models.ForeignKey(Consultation, on_delete=models.CASCADE)
-    # TODO - add more processing run metadata
-    started_at = models.DateTimeField(null=True)
-    finished_at = models.DateTimeField(null=True)
-    slug = models.SlugField(null=True)
+class ExecutionRun(UUIDPrimaryKeyModel, TimeStampedModel):
+    class TaskType(models.TextChoices):
+        SENTIMENT_ANALYSIS = "sentiment_analysis"
+        THEME_GENERATION = "theme_generation"
+        THEME_MAPPING = "theme_mapping"
 
-    @property
-    def themes(self):
-        return Theme.objects.filter(processing_run=self).distinct()
+    type = models.CharField(max_length=32, choices=TaskType.choices)
+    # TODO - add metadata e.g. langfuse_id
+    # Note, the execution run will be run on responses to a particular
+    # question part - but this will be stored in the correspondong framework/mapping etc.
+    history = HistoricalRecords()
```

Comment:
In which cases would these get edited? Do we need historical records here?

Reply:
Good challenge, we thought they might when we had SQL table convos but if I'm aiming to be minimal, the `HistoricalRecords` should come out and we can re-add if/when needed.

Reply:
`HistoricalRecords` have now been removed for the `ExecutionRun` model (https://github.com/i-dot-ai/consult/pull/405/commits/517f14391a6c9008a953ce9152e36ae3e20adb92).